packages/presence-signal/src/createPresenceSignal.ts

```diff
@@ -0,0 +1,108 @@
+import { createSignal, createEffect, onCleanup, type Accessor } from "solid-js";
+
+type SharedTransitionConfig = {
+  /** Duration in milliseconds used both for enter and exit transitions. */
+  transitionDuration: number;
+};
+
+type SeparateTransitionConfig = {
+  /** Duration in milliseconds used for enter transitions (overrides `transitionDuration` if provided). */
+  enterTransitionDuration: number;
+  /** Duration in milliseconds used for exit transitions (overrides `transitionDuration` if provided). */
+  exitTransitionDuration: number;
```

Comment:
Shouldn't these be optional?
Also why not just call it `duration`, `exitDuration`, and `enterDuration`? transition is the point of this primitive so it will implicitly involve transition
```suggestion
  enterTransitionDuration?: number;
  /** Duration in milliseconds used for exit transitions (overrides `transitionDuration` if provided). */
  exitTransitionDuration?: number;
```

Reply:
I do agree that `exitDuration` & `enterDuration` say everything that needs to be said, but I am concerned that `duration` might be too terse of an option property. Do you think it's obvious enough what it is a duration _of_?

Reply:
I have updated `exitDuration` & `enterDuration` seeing as we agree on that change.